src/clarity-icons/clarity-icons-element.ts

```diff
@@ -31,21 +37,28 @@ ClarityIconElement.prototype.constructor = ClarityIconElement;
 
 const replaceTitleWithCustom = function(currentShapeTemplate: string, currentTitleAttrVal: string,
                                         clrIconUniqId: string) {
-    const hasSVGtag = currentShapeTemplate.match(/<svg\b[^>]*>/i);  // will find the first occurrence of svg tag
+    const tempContainer = document.createElement("div");
 
-    if (!hasSVGtag) {
+    tempContainer.innerHTML = currentShapeTemplate;
+    const svgElement = tempContainer.querySelector("svg");
+
+    if (!svgElement) {
         return currentShapeTemplate;
     }
 
-    // if the title tag doesn't exist right after that first svg tag, we will inject a
-    if (currentShapeTemplate.indexOf("<title") > -1 && currentShapeTemplate.indexOf("</title>") > -1) {
-        // if title tag exists, remove that fist
-        currentShapeTemplate = currentShapeTemplate.replace(/<title[^>]*>([^<]+)<\/title>/i, "");
+    let titleElement = svgElement.querySelector("title");
+
+    if (titleElement) {
+        titleElement.setAttribute("id", clrIconUniqId);
+        titleElement.innerHTML = currentTitleAttrVal;
+    } else {
+        titleElement = document.createElement("title");
+        titleElement.setAttribute("id", clrIconUniqId);
+        titleElement.innerHTML = currentTitleAttrVal;
+        svgElement.insertAdjacentElement("afterbegin", titleElement);
     }
 
-    const openingSVGTag = hasSVGtag[0];
-    const remainingSVG = currentShapeTemplate.split(/<svg\b[^>]*>/i)[1];
-    return openingSVGTag + `<title id="${clrIconUniqId}">${currentTitleAttrVal}</title>` + remainingSVG;
+    return tempContainer.innerHTML;
```

Comment:
We should not be going from `String -> Html -> String -> Html`

I think we can do what we want with basic (and _fast_!) string operations and then apply it to the DOM once.

Reply:
@youdz @mathisscott I resorted to DOM operations as I thought this would be more robust and reliable than using regexes and string operations. When I do benchmark tests, this approach happened to show slightly better performance than the previous string manipulation approach I had. I think it doesn't perform badly because we are doing all the DOM operations **before** injecting it into the actual DOM tree. So there is no deep DOM traversal or no DOM reflow activity.  Also, in terms of readability, it looks much cleaner and easier to read IMO.

The only hesitation or question over this approach I have now is: can we accomplish better results if we find better string manipulating operations? I think better string manipulating operations means we should avoid regexes as much as possible because as soon as I use them, the benchmark tests are not showing better results, but actually worse. And manipulating string HTML template without regex to reach the desired result can be very difficult.

Reply:
Or we could also end up achieving better results if we use both dom operations and string manipulating operators. That's what I've been pondering, testing, and experimenting.

Reply:
I would say:

• avoid regex for this